data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/extension/MaxMindConfig.java

```diff
@@ -57,6 +68,19 @@ boolean isAwsAuthenticationOptionsRequired() {
         return true;
     }
 
+    @AssertTrue(message = "database_paths should be https endpoint if using URL and if insecure is set to false")
+    public boolean isSecureEndpoint() throws URISyntaxException {
```

Comment:
Please consider renaming this method. The logic below looks incorrect based on the name. Perhaps `isHttpsEndpointOrInsecure`?